src/net_processing.cpp

```diff
@@ -2052,7 +2052,9 @@ void PeerManager::ProcessOrphanTx(std::set<uint256>& orphan_work_set)
             break;
         }
     }
-    m_mempool.check(&::ChainstateActive().CoinsTip());
+    CChainState& active_chainstate = m_chainman.ActiveChainstate();
+    CCoinsViewCache& active_coins_tip = active_chainstate.CoinsTip();
+    m_mempool.check(&active_coins_tip, active_chainstate.m_blockman.GetSpendHeight(active_coins_tip));
```

Comment:
I don't follow what you mean by commit message "This is the only instance where validation reaches for something outside of it", you mean the mempool accessing `spendheight` ?

Reply:
I mean that in the codepath I'm touching, `CChainState::ActivateBestChainStep` (which resides in `validation.cpp`) is calling (reaching for) `CTXMemPool::Check` (which resides in `txmempool.cpp`), which means that I need to resolve `CTXMemPool::Check`'s reference of `g_chainman` first.